source/common/common/envoy_defaults.cc

```diff
@@ -0,0 +1,95 @@
+                            ProtobufMessage::getNullValidationVisitor());
+}
+
+const DefaultsProfile& DefaultsProfile::get() {
+  if (DefaultsProfileSingleton::getExisting()) {
+    return DefaultsProfileSingleton::get();
+  }
+
+  static DefaultsProfile no_singleton_profile_;
+  return no_singleton_profile_;
+};
+
+// Retrieve number value in defaults profile @ config_name.field if it exists, otherwise return
+// `default_value`
+double DefaultsProfile::get_number(const std::string config_name, const std::string& field,
+                                   double default_value) const {
+  absl::optional<ProtobufWkt::Value> value = get_proto_value(config_name, field);
+
+  if (value && value->has_number_value()) {
+    printf("NUMBER VALUE: %f\n\n", value->number_value());
+    return value->number_value();
+  }
+
+  return default_value;
+}
+
+// Retrieve ms value in defaults profile @ config_name.field if it exists, otherwise return
+// `default_value`
+std::chrono::milliseconds DefaultsProfile::get_ms(const std::string config_name,
+                                                  const std::string& field,
+                                                  int default_value) const {
+  absl::optional<ProtobufWkt::Value> value = get_proto_value(config_name, field);
+
+  if (value && value->has_string_value()) {
+    ProtobufWkt::Duration duration;
+    auto result = ProtobufUtil::JsonStringToMessage("\"" + value->string_value() + "\"", &duration);
+    // should I use TestUtility::loadFromJson("\"" + value->string_value() + "\"", &duration) ?
+    printf("DURATION VALUE: %lds\n\n", duration.seconds());
+    return std::chrono::milliseconds(result.ok() ? DurationUtil::durationToMilliseconds(duration)
+                                                 : default_value);
+  }
+
+  return std::chrono::milliseconds(default_value);
+}
+
+// Retrieve bool value in defaults profile @ config_name.field if it exists, otherwise return
+// `default_value`
+bool DefaultsProfile::get_bool(const std::string config_name, const std::string& field,
+                               bool default_value) const {
+  absl::optional<ProtobufWkt::Value> value = get_proto_value(config_name, field);
+
+  if (value && value->has_bool_value()) {
+    printf("BOOL VALUE: %s\n\n", value->bool_value() ? "true" : "false");
+    return value->bool_value();
+  }
+
+  return default_value;
+}
+
+// Search defaults profile for `field`, return value if found
+absl::optional<ProtobufWkt::Value>
+DefaultsProfile::get_proto_value(const std::string config_name, const std::string& field) const {
+  printf("\nFINDING %s.%s\n", config_name.c_str(), field.c_str());
+  auto fields_map = defaults_manifest_.fields();
+  auto config_it = fields_map.find(config_name);
+
+  // case: `field` exists in a map of multiple fields
+  if (config_it != fields_map.end()) {
+    const ProtobufWkt::Value submap = (config_it->second).edge_config().example();
```

Comment:
I think we are missing a case where `edge_config` doesn't have the `example` field present.